servicetalk-transport-api/src/main/java/io/servicetalk/transport/api/EarlyConnectionAcceptor.java

```diff
@@ -29,14 +29,32 @@ public interface EarlyConnectionAcceptor extends ExecutionStrategyInfluencer<Con
     /**
      * Accept or reject an incoming connection.
      * <p>
-     * Note that the {@link ConnectionInfo#sslSession()} will always return null with this acceptor since it is called
-     * before the TLS handshake is performed (and as a result no SSL session has been established).
+     * Note that the {@link ConnectionInfo#sslSession()} will always return {@code null} with this acceptor since it
+     * is called before the TLS handshake is performed (and as a result no SSL session has been established).
      *
-     * @param info additional information about the connection to make an acceptance decision.
-     * @return a completed (to accept) or a failed (to reject) {@link Completable}
+     * @param info {@link ConnectionInfo} to make an acceptance decision
+     * @return {@link Completable#completed()} to accept, or {@link Completable#failed(Throwable)} to reject with the
+     * cause
+     * @deprecated Implement {@link #accept(ConnectionContext)} instead
      */
+    @Deprecated // FIXME: 0.43 - swap default implementation with non-deprecated method
     Completable accept(ConnectionInfo info);
```

Comment:
This is why I dislike `@FunctionalInterface`s, they don't allow adding default impl for a method used as lambda, which requires 2 major releases to get rid of the deprecated method. Assuming all users implemented new overload, it means in 0.43 we can swap `default` impl`, and only in `0.44` we can remove the deprecated method 😞